Tests/ProtobufferCodingTests/ProtobufDeccoderTests.swift

```diff
@@ -0,0 +1,166 @@
+//
+//  ProtobufDecoderTests.swift
+//  
+//
+//  Created by Moritz Schüll on 26.11.20.
+//
+
+import Foundation
+import XCTest
+@testable import ProtobufferCoding
+
+
+class ProtobufDecoderTests: XCTestCase {
+
+    struct Message<T: Codable>: Codable {
+        var content: T
+        enum CodingKeys: Int, CodingKey {
+            case content = 1
+        }
+    }
+
+    struct ComplexMessage: Codable {
+        public var numberInt32: Int32
+        public var numberUint32: UInt32
+        public var numberBool: Bool
+        public var enumValue: Int32
+        public var numberDouble: Double
+        public var content: String
+        public var byteData: Data
+        public var nestedMessage: Message<String>
+        public var numberFloat: Float
+
+        enum CodingKeys: String, CodingKey, ProtoCodingKey {
+            case numberInt32
+            case numberUint32
+            case numberBool
+            case enumValue
+            case numberDouble
+            case content
+            case byteData
+            case nestedMessage
+            case numberFloat
+
+            static func mapCodingKey(_ key: CodingKey) throws -> Int? {
+                switch key {
+                case CodingKeys.numberInt32:
+                    return 1
+                case numberUint32:
+                    return 2
+                case numberBool:
+                    return 4
+                case enumValue:
+                    return 5
+                case numberDouble:
+                    return 8
+                case content:
+                    return 9
+                case byteData:
+                    return 10
+                case nestedMessage:
+                    return 11
+                case numberFloat:
+                    return 14
+                default:
+                    throw ProtoError.unknownCodingKey(key)
+                }
+            }
+        }
+    }
+
+    func testDecodeInt32() throws {
+        let positiveData = Data([8, 185, 96])
+        let positiveOriginal: Int32 = 12345
+        let negativeData = Data([8, 199, 159, 255, 255, 255, 255, 255, 255, 255, 1])
+        let negativeOriginal: Int32 = -12345
+
+        let message1 = try ProtoDecoder().decode(Message<Int32>.self, from: positiveData)
```

Comment:
The additional message indicates to me that this should be split into two testMethods.